samples/dnn/dasiamrpn_tracker.cpp

```diff
@@ -0,0 +1,486 @@
+
+#include <iostream>
+#include <cmath>
+
+#include <opencv2/dnn.hpp>
+#include <opencv2/imgproc.hpp>
+#include <opencv2/highgui.hpp>
+
+using namespace cv;
+using namespace cv::dnn;
+
+const char *keys =
+        "{ help     h  |   | Print help message. }"
+        "{ input    i  |   | Full path to input video folder, the specific camera index. (empty for camera 0) }"
+        "{ net         | ./dasiamrpn_model.onnx | Full path to onnx model of net.}"
+        "{ kernel_cls1 | ./dasiamrpn_kernel_cls1.onnx | Full path to onnx model of net.}"
+        "{ kernel_r1   | ./dasiamrpn_kernel_r1.onnx | Full path to onnx model of net.}"
+        "{ backend     | 0 | Choose one of computation backends: "
+                            "0: automatically (by default), "
+                            "1: Halide language (http://halide-lang.org/), "
+                            "2: Intel's Deep Learning Inference Engine (https://software.intel.com/openvino-toolkit), "
+                            "3: OpenCV implementation }"
+        "{ target      | 0 | Choose one of target computation devices: "
+                            "0: CPU target (by default), "
+                            "1: OpenCL, "
+                            "2: OpenCL fp16 (half-float precision), "
+                            "3: VPU }";
+
+// Initial parameters of the model
+struct trackerConfig
+{
+    float windowInfluence = 0.43f;
+    float lr = 0.4f;
+    int scale = 8;
+    bool swapRB = false;
+    int totalStride = 8;
+    float penaltyK = 0.055f;
+    int exemplarSize = 127;
+    int instanceSize = 271;
+    float contextAmount = 0.5f;
+    std::vector<float> ratios = {0.33f, 0.5f, 1.0f, 2.0f, 3.0f};
+    int anchorNum = int(ratios.size());
+    Mat anchors;
+    Mat windows;
+    Scalar avgChans;
+    Size imgSize = {0,0};
+    Rect2f targetBox = {0,0,0,0};
+    int scoreSize = (instanceSize-exemplarSize)/totalStride+1;
+
+    void update_scoreSize()
+    {
+        scoreSize = int((instanceSize - exemplarSize) / totalStride + 1);
+    }
+};
+
+void softmax(const Mat &src, Mat & dst);
+void elementMax(Mat & src);
+Mat generateHanningWindow(const trackerConfig &trackState);
+Mat generateAnchors(trackerConfig &trackState);
+Mat getSubwindow(Mat &img, const Rect2f &targetBox, float originalSize, Scalar avgChans);
+float trackerEval(Mat img, trackerConfig &trackState, Net &siamRPN);
+void trackerInit(Mat img, trackerConfig &trackState, Net &siamRPN, Net &siamKernelR1, Net &siamKernelCL1);
+
+template<typename T>
+T sizeCal(const T &w, const T &h)
+{
+    T pad = (w + h)*T(0.5);
+    T sz2 = (w + pad) * (h + pad);
+    return sqrt(sz2);
+}
+
+template<>
+Mat sizeCal(const Mat &w, const Mat &h)
+{
+    Mat pad = (w + h)*0.5;
+    Mat sz2 = (w + pad).mul((h + pad));
+
+    cv::sqrt(sz2, sz2);
+    return sz2;
+}
+
+int main(int argc, char **argv)
+{
+    // Parse command line arguments.
+    CommandLineParser parser(argc, argv, keys);
+
+    if (argc == 1 || parser.has("help"))
```

Comment:
It looks strange. All parameters have default values and I expect that sample should work by default with camera 0, if all models are there.